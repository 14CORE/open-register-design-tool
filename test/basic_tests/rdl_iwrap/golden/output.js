/*   Ordt 170413.03 autogenerated file 
 *   Input: ./rdl_iwrap/test.rdl
 *   Parms: ./rdl_iwrap/test.parms
 *   Date: Thu Apr 13 23:28:53 EDT 2017
 */

register_set top "Registers for top" {

  address = 0x0;
  register_width = 32;
  register_set regs_none "regs_none register_set" {
    address = 0x0;
    
    register reg_none "reg_none register" {
      address = 0x0;
      nop[28];
      integer fld_int[1] "fld_int field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_ext[1] "fld_ext field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_gen[1] "fld_gen field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_none[1] "fld_none field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register reg_int "reg_int register" {
      address = 0x4;
      nop[28];
      integer fld_int[1] "fld_int field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_ext[1] "fld_ext field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_gen[1] "fld_gen field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_none[1] "fld_none field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x8;
  };
  
  register_set regs_ext "regs_ext register_set" {
    address = 0x8;
    
    register reg_none "reg_none register" {
      address = 0x0;
      nop[28];
      integer fld_int[1] "fld_int field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_ext[1] "fld_ext field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_gen[1] "fld_gen field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_none[1] "fld_none field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register reg_int "reg_int register" {
      address = 0x4;
      nop[28];
      integer fld_int[1] "fld_int field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_ext[1] "fld_ext field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_gen[1] "fld_gen field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer fld_none[1] "fld_none field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x8;
  };
  
  register_set_size = 0x10;
};

