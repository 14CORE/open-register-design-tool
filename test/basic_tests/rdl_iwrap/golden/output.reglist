//   Ordt 160822.01 autogenerated file 
//   Input: ./rdl_iwrap/test.rdl
//   Parms: ./rdl_iwrap/test.parms
//   Date: Thu Aug 25 21:09:34 EDT 2016
//

0x0              size: 32   fields: 4    bits: 4         regs_none.reg_none
0x4              size: 32   fields: 4    bits: 4         regs_none.reg_int
0x8              size: 32   fields: 4    bits: 4         regs_ext.reg_none
0xc              size: 32   fields: 4    bits: 4         regs_ext.reg_int
