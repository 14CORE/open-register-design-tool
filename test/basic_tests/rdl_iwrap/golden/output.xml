<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 160822.01 autogenerated file 
       Input: ./rdl_iwrap/test.rdl
       Parms: ./rdl_iwrap/test.parms
       Date: Thu Aug 25 21:09:34 EDT 2016
 -->

<map version="160822.01">
  <id>top</id>
  <baseaddr>0x0</baseaddr>
  <shorttext>top registers</shorttext>
  <regset>
    <id>regs_none</id>
    <shorttext>regs_none registers</shorttext>
    <baseaddr>0x0</baseaddr>
    <reg>
      <id>reg_none</id>
      <parentpath>regs_none</parentpath>
      <shorttext>reg_none register</shorttext>
      <baseaddr>0x0</baseaddr>
      <width>32</width>
      <field>
        <id>fld_none</id>
        <shorttext>fld_none field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_gen</id>
        <shorttext>fld_gen field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_ext</id>
        <shorttext>fld_ext field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>2</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_int</id>
        <shorttext>fld_int field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>3</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>reg_int</id>
      <parentpath>regs_none</parentpath>
      <shorttext>reg_int register</shorttext>
      <baseaddr>0x4</baseaddr>
      <width>32</width>
      <field>
        <id>fld_none</id>
        <shorttext>fld_none field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_gen</id>
        <shorttext>fld_gen field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_ext</id>
        <shorttext>fld_ext field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>2</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_int</id>
        <shorttext>fld_int field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>3</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x7</highaddr>
  </regset>
  <regset>
    <id>regs_ext</id>
    <shorttext>regs_ext registers</shorttext>
    <baseaddr>0x8</baseaddr>
    <reg>
      <id>reg_none</id>
      <parentpath>regs_ext</parentpath>
      <shorttext>reg_none register</shorttext>
      <baseaddr>0x8</baseaddr>
      <width>32</width>
      <field>
        <id>fld_none</id>
        <shorttext>fld_none field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_gen</id>
        <shorttext>fld_gen field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_ext</id>
        <shorttext>fld_ext field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>2</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_int</id>
        <shorttext>fld_int field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>3</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>reg_int</id>
      <parentpath>regs_ext</parentpath>
      <shorttext>reg_int register</shorttext>
      <baseaddr>0xc</baseaddr>
      <width>32</width>
      <field>
        <id>fld_none</id>
        <shorttext>fld_none field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_gen</id>
        <shorttext>fld_gen field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_ext</id>
        <shorttext>fld_ext field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>2</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld_int</id>
        <shorttext>fld_int field</shorttext>
        <access>RW</access>
        <reset>0</reset>
        <lowidx>3</lowidx>
        <width>1</width>
        <hwmod></hwmod>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0xf</highaddr>
  </regset>
</map>
